### ISSUE TITLE
Cannot assign SlimLoadingBarComponent to directives

### ISSUE THREAD

**Issue by BenDevelopment** [akserg/ng2-slim-loading-bar#16]
When I try to set up ng2-slim-loading-bar, I got an error in this line :
`directives: [SlimLoadingBarComponent],`
The error is :
`The argument of type "{guidelines: typeof SlimLoadingBar Component [];}" is not attributable to the type of parameter 'ComponentMetadataType'.`
I'm using angular2 rc6. What am I doing wrong ?

**Comment by akserg** [comment]
Hi @BenDevelopment,

I migrate the Slim Loading Bar to Angular 2.0.0. May I ask you to try it. 
Please follow the instructions I updated in [Usage](https://github.com/akserg/ng2-slim-loading-bar#usage) section of README.

Let me know if doesn't work.

**Comment by BenDevelopment** [comment]
Thant you @akserg for your answer.
I have followed the instructions, but I still have a problem:
`(index):30 Error: Error: Unexpected value 'SlimLoadingBarService' exported by the module 'SlimLoadingBarModule'(…)`

**Comment by akserg** [comment]
Do you use Angular 2.0.0 or Angular 2.0.0-RC6?

**Comment by BenDevelopment** [comment]
2.0.0 (I have downgraded since my first post)

**Comment by akserg** [comment]
Do you mean 2.0.0 release?

**Comment by BenDevelopment** [comment]
Yes.
Here is my package.json, maybe it can help:
```
{
  "name": "kettelbell-me",
  "version": "1.0.0",
  "description": "Kettelbell.me. Social crossfit app",
  "scripts": {
    "start": "tsc && concurrently \"tsc -w\" \"lite-server\" ",
    "docker-build": "docker build -t ng2-quickstart .",
    "docker": "npm run docker-build && docker run -it --rm -p 3000:3000 -p 3001:3001 ng2-quickstart",
    "pree2e": "npm run webdriver:update",
    "e2e": "tsc && concurrently \"http-server -s\" \"protractor protractor.config.js\" --kill-others --success first",
    "lint": "tslint ./app/**/*.ts -t verbose",
    "lite": "lite-server",
    "postinstall": "typings install",
    "test": "tsc && concurrently \"tsc -w\" \"karma start karma.conf.js\"",
    "test-once": "tsc && karma start karma.conf.js --single-run",
    "tsc": "tsc",
    "tsc:w": "tsc -w",
    "typings": "typings",
    "webdriver:update": "webdriver-manager update"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@angular/common": "2.0.0",
    "@angular/compiler": "2.0.0",
    "@angular/core": "2.0.0",
    "@angular/forms": "2.0.0",
    "@angular/http": "2.0.0",
    "@angular/platform-browser": "2.0.0",
    "@angular/platform-browser-dynamic": "2.0.0",
    "@angular/router": "3.0.0",
    "@angular/upgrade": "2.0.0",
    "angular2-in-memory-web-api": "0.0.20",
    "bootstrap": "^3.3.6",
    "core-js": "^2.4.1",
    "ng2-slim-loading-bar": "^2.0.0",
    "reflect-metadata": "^0.1.3",
    "rxjs": "5.0.0-beta.12",
    "systemjs": "0.19.27",
    "zone.js": "^0.6.23"
  },
  "devDependencies": {
    "concurrently": "^2.2.0",
    "lite-server": "^2.2.0",
    "typescript": "^2.0.2",
    "typings": "^1.0.4",
    "canonical-path": "0.0.2",
    "http-server": "^0.9.0",
    "tslint": "^3.7.4",
    "lodash": "^4.11.1",
    "jasmine-core": "~2.4.1",
    "karma": "^1.2.0",
    "karma-chrome-launcher": "^0.2.3",
    "karma-cli": "^0.1.2",
    "karma-htmlfile-reporter": "^0.2.2",
    "karma-jasmine": "^0.3.8",
    "protractor": "^3.3.0",
    "rimraf": "^2.5.2"
  },
  "repository": {}
}

```

**Comment by akserg** [comment]
Sorry, I'm asking because you send downgraded from RC6. I assume you upgraded. By the way - any source code (may be small portion)?

**Comment by BenDevelopment** [comment]
Yes sorry I get mixed up. ^^ 

Here are some files:
systemjs.config.js:
```
/**
 * System configuration for Angular 2 samples
 * Adjust as necessary for your application needs.
 */
(function (global) {
  System.config({
    paths: {
      // paths serve as alias
      'npm:': 'node_modules/'
    },
    // map tells the System loader where to look for things
    map: {
      // our app is within the app folder
      app: 'app',

      // angular bundles
      '@angular/core': 'npm:@angular/core/bundles/core.umd.js',
      '@angular/common': 'npm:@angular/common/bundles/common.umd.js',
      '@angular/compiler': 'npm:@angular/compiler/bundles/compiler.umd.js',
      '@angular/platform-browser': 'npm:@angular/platform-browser/bundles/platform-browser.umd.js',
      '@angular/platform-browser-dynamic': 'npm:@angular/platform-browser-dynamic/bundles/platform-browser-dynamic.umd.js',
      '@angular/http': 'npm:@angular/http/bundles/http.umd.js',
      '@angular/router': 'npm:@angular/router/bundles/router.umd.js',
      '@angular/forms': 'npm:@angular/forms/bundles/forms.umd.js',

      // other libraries
      'rxjs':                       'npm:rxjs',
      'angular2-in-memory-web-api': 'npm:angular2-in-memory-web-api',
      'ng2-slim-loading-bar': 'node_modules/ng2-slim-loading-bar',
    },
    // packages tells the System loader how to load when no filename and/or no extension
    packages: {
      app: {
        main: './main.js',
        defaultExtension: 'js'
      },
      rxjs: {
        defaultExtension: 'js'
      },
      'angular2-in-memory-web-api': {
        main: './index.js',
        defaultExtension: 'js'
      },
      'ng2-slim-loading-bar': { main: 'index.js', defaultExtension: 'js' },
    }
  });
})(this);

```

app.module.ts:
```
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { HttpModule } from '@angular/http';
import { routing } from './app.routing';
import { HashLocationStrategy, LocationStrategy } from '@angular/common';
import { FormsModule } from '@angular/forms';

import { SlimLoadingBarModule } from 'ng2-slim-loading-bar';

import { AppComponent } from './app.component';
import { LoginComponent } from './login/index';
import { HomeComponent } from './home/index';
import { CoachingRequestComponent } from './coaching_request/index';
import { AuthGuard } from './_guards/index';
import { AuthService } from './_services/index';
import { CoachingRequestService } from './coaching_request/coaching_request.service';

@NgModule({
    imports: [
        BrowserModule,
        FormsModule,
        HttpModule,
        routing,
        SlimLoadingBarModule,
    ],
    declarations: [
        AppComponent,
        LoginComponent,
        HomeComponent,
        CoachingRequestComponent
    ],
    providers: [
        { provide: LocationStrategy, useClass: HashLocationStrategy },
        AuthGuard,
        AuthService,
        CoachingRequestService,
    ],
    bootstrap: [AppComponent]
})
export class AppModule { }

```
app.component.ts:
```
import { Component, OnInit, OnDestroy } from '@angular/core';
import { AuthUser } from './_models/index';
import { AuthService } from './_services/index';
import { SlimLoadingBarService } from 'ng2-slim-loading-bar';

@Component({
    moduleId: module.id,
    selector: 'my-app',
    templateUrl: 'app.component.html'
})

export class AppComponent implements OnInit, OnDestroy {
    user: AuthUser;
    private loginSub: any;

    constructor(private authService: AuthService, private slimLoadingBarService: SlimLoadingBarService) {
        this.user = authService.getLoggedUser();
    }

    ngOnInit() {
        this.loginSub = this.authService.loggedInObservable.subscribe(val => {
            this.user = val;
        });
    }

    ngOnDestroy() {
        this.loginSub.unsubscribe();
    }

    startLoading() {
        // We can listen when loading will be completed
        this.slimLoadingBarService.start(() => {
            console.log('Loading complete');
        });
    }

}

```
app.component.html:
```
<button (click)="startLoading()">Start Loading</button>
<ng2-slim-loading-bar></ng2-slim-loading-bar>
```
I added style.css in my index.html's head:
`        <link rel="stylesheet" href="node_modules/ng2-slim-loading-bar/style.css">`

**Comment by akserg** [comment]
Ok, let me see.

**Comment by akserg** [comment]
Fixed in 2.0.1. Check README for updated docs how to use.
